src/radical/pilot/raptor/worker.py

```diff
@@ -481,12 +482,14 @@ def _dispatch_proc(self, task):
             uid  = task['uid']
             exe  = task['description']['executable']
             args = task['description'].get('arguments', list())
-            tenv = task['description'].get('environment', dict())
+            env  = ru.dict_merge(self._task_env,
```

Comment:
This is effectively the same as `self._task_env.update(task['description']['environment'])`, right?

Reply:
Yes indeed.  Well, since env dicts are not hierarchical we could use that in fact - brain was on autotyping mode...